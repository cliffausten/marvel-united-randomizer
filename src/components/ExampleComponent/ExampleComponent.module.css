.example-component {
  display: grid;
  max-width: 700px;
  grid-gap: 10px;
  justify-items: center;
}

.quote {
  text-align: center;
  font-style: italic;
}

.author {
  font-weight: bold;
}
