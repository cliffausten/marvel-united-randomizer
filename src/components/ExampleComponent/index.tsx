import React, { useEffect } from 'react';
import useStores from 'hooks/useStores';
import { observer } from 'mobx-react';

import styles from './ExampleComponent.module.css';

type Props = {
  showAuthor: boolean;
};

const ExampleComponent = observer(({ showAuthor }: Props) => {
  const { exampleStore } = useStores();

  useEffect(() => {
    exampleStore.getNewQuote();
  });

  return (
    <div className={styles['example-component']}>
      <div className={styles['quote']}>{exampleStore.quote}</div>
      {showAuthor && (
        <div className={styles['author']}>{exampleStore.author}</div>
      )}
    </div>
  );
});

export default ExampleComponent;
