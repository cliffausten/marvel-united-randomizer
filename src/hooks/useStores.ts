import React from 'react';
import { storeContext } from 'components/StoreProvider';
import RootStore from 'stores/RootStore';

export const useStores = (): RootStore => {
  const store = React.useContext(storeContext);
  if (!store) {
    throw new Error('useStores must be used within a StoreProvider.');
  }
  return store;
};

export default useStores;
