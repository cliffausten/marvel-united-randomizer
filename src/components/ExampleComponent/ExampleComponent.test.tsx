import React from 'react';

import ExampleComponent from './index';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import useStores from 'hooks/useStores';
jest.mock('hooks/useStores');

describe('ExampleComponent', () => {
  it('Render', () => {
    (useStores as jest.Mock).mockImplementation(() => {
      return {
        exampleStore: {
          quote: 'TEST_QUOTE',
          author: 'TEST_AUTHOR'
        }
      };
    });

    const exampleComponent = shallow(<ExampleComponent />);

    expect(exampleComponent).toMatchSnapshot();
  });

  it('useEffect', () => {
    const getQuoteMock = jest.fn();

    (useStores as jest.Mock).mockImplementation(() => {
      return {
        exampleStore: {
          getNewQuote: getQuoteMock
        }
      };
    });

    act(() => {
      mount(<ExampleComponent />);
    });

    expect(getQuoteMock.mock.calls.length).toBe(1);
  });
});
