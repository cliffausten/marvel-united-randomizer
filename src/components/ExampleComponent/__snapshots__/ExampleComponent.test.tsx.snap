// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent Render - showAuthor = false 1`] = `
<div
  className="example-component"
>
  <div
    className="quote"
  >
    TEST_QUOTE
  </div>
</div>
`;

exports[`ExampleComponent Render - showAuthor = true 1`] = `
<div
  className="example-component"
>
  <div
    className="quote"
  >
    TEST_QUOTE
  </div>
  <div
    className="author"
  >
    TEST_AUTHOR
  </div>
</div>
`;
